example/broker.cpp

```diff
@@ -16,18 +16,43 @@
 #include <fstream>
 
 #if defined(MQTT_USE_TLS)
-boost::asio::ssl::context init_ctx(boost::program_options::variables_map const& vm)
+
+template<typename Server>
+void reload_ctx(boost::asio::steady_timer &reload_timer, Server &server, boost::program_options::variables_map const& vm, char const *name)
```

Comment:
```suggestion
void reload_ctx(boost::asio::steady_timer& reload_timer, Server& server, boost::program_options::variables_map const& vm, char const* name)
```

Reply:
Passing `boost::program_options::variables_map const& vm` is not good.
Analyze program options is not suitable for reload_ctx responsibility.
So certificate, private_key, and certificate_reload_interval should be passed.
The arguments combination checking should be done at run_broker().

Reply:
This still needs to be updated ?

Reply:
No, the function itself already updated as different parameters.

Reply:
Yes, I updated. Binding the variables_map to the timer is not a good idea.

Reply:
> No, the function itself already updated as different parameters.

It is wrong. The answer is yes. I will add comments.

Reply:
I god confused between load_ctx and reload_ctx. I need a time to understand.

Reply:
ok

Reply:
Now I understand. No updated is required here. reload_ctx() 's parameter is OK.

Reply:
only coding rules update ? 

passing as const & does result in copy everytime the callback timer is called